### PR TITLE
[21856] Unload Plugin stack from memory

### PR DESCRIPTION
**Pull request by livecodepanos**
See https://quality.livecode.com/show_bug.cgi?id=21856 for a detailed description of the problem

### CODE REVIEW

File: Toolset/palettes/start center/revStartCenterBehavior.livecodescript [hunk 1]
```diff
@@ -104,11 +104,13 @@ on ideOpenStack pTarget
       if tLine ends with ".livecode" or tLine ends with ".livecodescript" or \
             tLine ends with ".rev" then 
          put the substacks of stack tLine into tSubstacks
+         // Unload Plugin from memory
+         delete stack tLine
```
Comment (trevordevore) [inline Toolset/palettes/start center/revStartCenterBehavior.livecodescript]: @livecodepanos Will this change IDE behavior if you always delete the plugin stack, even if it was already in memory?

Comment (livecodepanos) [reply Toolset/palettes/start center/revStartCenterBehavior.livecodescript]: @trevordevore hmm good point. If the stack is already in memory but not open, then I believe this change will not have any other unwanted consequences. 

But if the stack is already open, then this line will cause the stack to close, which is something we don't want to. I guess we should do something like:

```
if not (the short name of stack tLine is in the openstacks) then
   delete stack tLine
end if
```

Comment (trevordevore) [reply Toolset/palettes/start center/revStartCenterBehavior.livecodescript]: @livecodepanos This is better. Ideally it wouldn't alter the state of the stack at all (if the stack is among the mainstacks it should remain among the mainstacks) but that isn't possible with existing syntax. Purging a stack from memory can affect script local variables. `the stacks` returns stack filenames that have open stacks but there is no equivalent for filenames of `mainstacks`.

**Comment by runrevmark** [comment]
@livecodepanos (and @trevordevore so he can make sure I'm not talking rubbish!) Wouldn't this work:
```
if the filename of stack (the mainstack of pTarget) is among the lines of revIDEGetAllPlugins() then
  exit ...
end if
```
It would avoid loading any stacks into memory at all.

**Comment by trevordevore** [comment]
@runrevmark - From what I can tell that code wouldn't load any additional stacks into memory. I didn't look to see if it accomplishes the same goal. 

As a side note, I was longing for a `stacks`-like property the other day that would return the filenames of all `mainstacks`.

**Comment by livecodepanos** [comment]
Nice one @runrevmark . I will slightly modify this snippet, as `pTarget` is not necessarily a stack (might be a card) so `the mainstack of pTarget` will throw an error.

**Comment by runrevmark** [comment]
@livecode-vulcan review ok 9beee9c

**Comment by livecode-vulcan** [comment]
💙 review by @runrevmark ok 9beee9c970262fb72ae705c00f6b28a0feb98c68

**Pull request merged**